File: 7kyu 1/findShort.js
function findShort(s) {
   const arr = s.split(" ");
   let min = arr[0].length;
   for (let i = 0; i < arr.length; i++) {
      let hsl;
      for (let j = 0; j < arr[i].length; j++) {
         hsl = arr[i].length;
      }
      if (hsl < min) min = hsl;
   }
   return min;
}


function findShort(s) {
   const arr = s.split(' ');
   let smallest = arr[0];
   for (let i = 0; i < arr.length; i++) {
      if (arr[i].length < smallest.length) {
         smallest = arr[i];
      }
   }
   return smallest.length;
}


const findShort = s => Math.min.apply(null, s.split(' ').map(w => w.length));


const findShort = (s) => s.split(' ').sort((a, b) => b.length - a.length).pop().length;


const findShort = s => s.split(' ').map(a => a.length).reduce((a, b) => Math.min(a, b));


console.log(findShort("sulton muhajir"), 6);
console.log(findShort("bitcoin take over the world maybe who knows perhaps"), 3);
console.log(findShort("turns out random test cases are easier than writing out basic ones"), 3);

File: Python/7py 1/find_short.py
"""
|--------------------------------------------------------------------------
| Shortest Word
|--------------------------------------------------------------------------
|
"""

def find_short(s):
   arr = s.split(' ')
   minn = len(arr[0])
   for i in range(len(arr)):
      for j in range(len(arr[i])):
         res = len(arr[i])
      if res < minn:
         minn = res
   return minn


def find_short(s):
   return min(len(x) for x in s.split())


def find_short(s):
   return len(min(s.split(' '), key=len))


def find_short(s):
   return min(map(len, s.split(' ')))


def find_short(s):
   return len(sorted(s.split(), key=len)[0])


print(find_short("bitcoin take over the world maybe who knows perhaps"), 3)
print(find_short("turns out random test cases are easier than writing out basic ones"), 3)
print(find_short("lets talk about javascript the best language"), 3)
print(find_short("i want to travel the world writing code one day"), 1)
print(find_short("Lets all go on holiday somewhere very cold"), 2)   
print(find_short("Let's travel abroad shall we"), 2)